src/stores/ShardDetail.ts

```diff
@@ -85,74 +237,114 @@ export const getInitialState = (
                 'Shard List Set'
             );
         },
-        getShardStatusColor: (catalogNamespace) => {
-            const { shards } = get();
+        getTaskShards: (catalogNamespace, shards) => {
+            return shards.length > 0
+                ? shards.filter(({ spec }) => {
+                      const labels = spec.labels ? spec.labels.labels : [];
 
-            if (shards.length > 0) {
-                const selectedShard = shards.find(({ spec }) =>
-                    spec.id ? spec.id.includes(catalogNamespace) : undefined
+                      const taskName = labels?.find(
+                          (label) => label.name === 'estuary.dev/task-name'
+                      )?.value;
+
+                      return taskName === catalogNamespace;
+                  })
+                : [];
+        },
+        getTaskShardDetails: (taskShards) => {
+            if (taskShards.length > 0) {
+                const statusIndicators = taskShards.map((shard) =>
+                    evaluateTaskShardStatus(shard)
                 );
 
-                return selectedShard
-                    ? evaluateShardStatus(selectedShard)
-                    : defaultStatusColor;
+                return statusIndicators.length > 0
+                    ? statusIndicators
+                    : [defaultTaskShardDetail];
+            } else {
+                return [defaultTaskShardDetail];
+            }
+        },
+        getTaskStatusColor: (taskShardDetails) => {
```

Comment:
`getTaskStatusColor` returns the hex color code associated with the aggregate shard status for a specific task. As indicated by the name of its input argument, only the byproduct of `getTaskShardDetails` should be consumed by this function. Its only use is within the `EntityStatus` component to determine the color of the main status indicator.